CONTRIBUTING.md

```diff
@@ -1,108 +1,170 @@
+
+##### `install.sh`
+This is an installer script that:
+- Moves all the JARs to `$HOME/.terra/lib`
+- Moves the `terra` run script and the `README.md` file outside the unarchived install package directory
+- Deletes the unarchived install package directory
+- Sets the Docker image id to the default
+- Pulls the default Docker image id
+
+It is included in the `terra-cli.tar` install package in each GitHub release.
+It needs to be run from the same directory: `./install.sh`
+
+##### `terra`
+This is the run script that wraps the Java call to the CLI.
+- It looks for the JARs on the classpath in the `$HOME/.terra/lib` directory.
+- This script is generated by the Gradle application plugin, so any changes should be made there.
+
+It is included in the `terra-cli.tar` install package in each GitHub release.
+This is the script users can add to their `$PATH` to invoke the CLI more easily from another directory.
+
+
+### Docker
+The `docker/` directory contains files required to build the Docker image.
+All files in the `scripts/` sub-directory are copied to the image, into a sub-directory that is on the `$PATH`, 
+and made executable.
 
 #### Pull an existing image
-The `tools/local-dev.sh` script pulls the default image already.
+The `tools/local-dev.sh` and `install.sh` scripts pull the default image already. So this is mostly useful for
+development and debugging.
 
 The gcr.io/terra-cli-dev registry is public readable, so anyone should be able to pull images.
 
 To use a specific Docker image from GCR:
 1. Pull the image with that tag.
     ```
-    > ./gradlew pullDockerImage -PdockerImageTag=b5fdce0
-    Task :pullDockerImage
-    b5fdce0: Pulling from terra-cli-dev/terra-cli/v0.0
-    Digest: sha256:c77ee0b87a8972ec2a9f1b69387216a9f726f5503679edab37911a6322876dbe
-    Status: Downloaded newer image for gcr.io/terra-cli-dev/terra-cli/v0.0:b5fdce0
-    gcr.io/terra-cli-dev/terra-cli/v0.0:b5fdce0
-    
-    BUILD SUCCESSFUL in 3s
-    1 actionable task: 1 executed
+    > docker pull gcr.io/terra-cli-dev/terra-cli/v0.0:b5fdce0
     ```
-2. Update the image id that the CLI uses. (See output of previous command for image name and tag.)
+2. Update the image id that the CLI uses.
     ```
     > terra app set-image --image=gcr.io/terra-cli-dev/terra-cli/v0.0:b5fdce0
     ```
 
-#### Publish a new image
-The Gradle publishDockerImage task expects the caller to be logged into Vault already, because it fetches
-a SA key file in order to write to GCR. In the future, we should probably do this with a GH action instead.
+#### Build a new image
+For any change in the `docker/` directory to take effect:
+1. Build a new image. This uses a short Git hash for the current commit as the tag. See the script comments for more
+options.
+    ```
+    > ./tools/build-docker.sh
+   
+    Generating an image tag from the Git commit hash
+    Building the image
+    [...Docker output...]
+    Successfully built 6558c3bcb316
```

Comment:
What is this tag? Maybe add "docker image hash" or something for clarity.

Reply:
Yes, it's the Docker image id. But this line is actually part of the Docker output, so I can't modify it.